pkg/storage/chunk/client/aws/s3_storage_client.go

```diff
@@ -393,7 +393,7 @@ func (a *S3ObjectClient) GetObject(ctx context.Context, objectKey string) (io.Re
 	// Map the key into a bucket
 	bucket := a.bucketFromKey(objectKey)
 
-	var lastErr error
+	lastErr := ctx.Err()
```

Comment:
This is what the IBMCloud client does too, though I'm worried there's still an edge case where the context gets canceled in between this line and the first call to `retries.Ongoing()`. It sounds pretty unlikely so I didn't handle that here.